HMS-9277,HMS-5271: use DataView in the Repositories table

katarinazaprazna opened the pull request:
## Summary

This PR introduces a refactor of the Repositories table, switching its implementation to leverage the [DataView library from Patternfly](https://www.patternfly.org/extensions/data-view/overview). The goal is to align with best practices.

- The Bulk Select feature has been updated to use the [correct UX pattern](https://docs.google.com/document/d/1Wk6jH_wjfM2N9jIHft0Qa2lClijsMJ6AbWyFlJ7IKdE/edit?tab=t.fizfebrp68r8#heading=h.vsfcmwi5d26a), giving it a new look.

> We are currently enhancing the UX of the table by conditionally disabling the Bulk Select component. The current implementation for this control is not ideal. A request has been submitted to the PatternFly team to integrate the necessary disabling logic directly into the component. See the PatternFly ticket here: https://github.com/patternfly/react-component-groups/issues/819

- For consistency with DataView's naming conventions, the `searchQuery` variable was renamed to `search`.
- Custom styles were replaced with DataView's built-in styles, which is a recommendation from Patternfly.
- Updated the Kebab Menu copy to read "Delete repository" instead of "Delete 1 repositories" when only one repository is selected.
- Previously, users could select read-only repositories, and the kebab menu prevented them from being deleted. To improve the UX, we're now disabling the entire row for these repositories.

<img width="2239" height="1399" alt="Screenshot From 2025-09-22 18-50-42" src="https://github.com/user-attachments/assets/ae91fac2-7bee-43b1-956a-3879763c31e7" />


## Testing steps

- Verify that the app builds without any errors.
- Ensure that all automated tests pass.

We're now improving the UX by making read-only repository rows unselectable.

-  When a user selects a mix of read-only and custom repositories, the checkboxes for read-only rows will be disabled.
-  If a page contains only read-only repositories, the bulk select checkbox and the delete option will be disabled. A tooltip will inform the user, "Red Hat and EPEL repositories cannot be deleted."
- If a page has at least one custom repository, users will be able to delete it.

Besides disabling read-only rows, the table's behavior will remain the same. Users can only select all repositories on a single page at a time. A new pull request will be created to allow selection across multiple pages.

jlsherrill commented:
https://issues.redhat.com/browse/HMS-9277

xbhouse commented on src/Pages/Repositories/ContentListTable/ContentListTable.tsx:
```diff
@@ -392,138 +484,287 @@ const ContentListTable = () => {
-      return 'Snapshotting is in progress';
-    } else if (introspectStatus === 'Pending') {
-      return 'Introspection is in progress';
+  // Synchronize selection with the origin filter via targeted deselection (non-destructive)
```
yep that sounds good to me!

xbhouse replied:
i think a tooltip on either disabled checkbox or just on the bulk select in the toolbar when it's disabled could also be helpful. what do you think?

katarinazaprazna replied:
I agree with all your suggestions and will look into the DataView possibilities right away :)

jlsherrill commented:
https://issues.redhat.com/browse/HMS-5271

katarinazaprazna commented:
Hello @xbhouse  and @dominikvagner, I've addressed the issues you mentioned in your reviews. Please take a look at the improvements when you have a moment. Thanks!

katarinazaprazna commented:
/retest

katarinazaprazna commented:
Thanks for the investigation @xbhouse ❤️ 

I've managed to make it pass locally (it was sometimes passing before, too) by making a small tweak: ensuring we specifically target the kebab menu associated with the row. I'm pushing a commit now to see if this makes a difference in the pipeline run. Fingers crossed! 🤞

katarinazaprazna commented:
/retest

katarinazaprazna commented:
Hello @xbhouse and @dominikvagner, I've integrated the UX feedback from Maria into the code. Could you please take a look when you have a moment? Many thanks!

xbhouse reviewed (commented):
this looks good to me! rerunning the playwright tests a couple times to see if they're still flaky. if they are, i think we could add a retry to the kebab + menu item click actions

dominikvagner commented:
just noticed one potential issue with the `CustomRepoCRUD` test, while going through the traces 🔍 
```ts
await test.step('Confirm repo was updated', async () => {
  await page.getByRole('searchbox', { name: 'Filter by name/url' }).fill(`${repoName}-Edited`);
  const row = await getRowByNameOrUrl(page, `${repoName}-Edited`);
  ...
```
it shouldn't be doing this 🙅🏼 
it shouldn't be filling the searchbox, and then using the `getRow` helper as that will do the searching if needed 🐛 
I think this is a case of Playwright actually being tooo fast, _and/or the DataView auto search (searching on typing into the textbox) too slow_ 🤔 

in the first trace from [this](https://app.currents.dev/instance/ysBSqpJtu0DFbFNs) run and test, you can see the issue quite clearly 👀
1. it inputs stuff into the searchbox
2. gets the row (that helper) with the needed repo (it's on the first page, so the helper doesn't search for it) 
3. it tries to click the kebab toggle, but fails here because the auto search filters and reloads the table and that interrupts the opening of the menu 🕵🏼 

I think if we get rid of this issue, i.e. delete all instances of searching before using the get row helper, it should help/resolve the failures and flakiness 🙏🏼😅

katarinazaprazna commented:
@dominikvagner thanks so much for the help with this! I totally missed that issue. I've implemented your suggestions, and the tests are now finally passing locally on my end 🎉
